plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestCorruptedIcebergTable.java

```diff
@@ -0,0 +1,139 @@
+        assertQueryFailsIncorrectly("INSERT INTO " + tableName + " VALUES (NULL, NULL, ROW(NULL, NULL))", "Failed to open input stream for file: .*");
+        assertQueryFailsIncorrectly("UPDATE " + tableName + " SET country = 'AUSTRIA'", "Failed to open input stream for file: .*");
+        assertQueryFailsIncorrectly("DELETE FROM " + tableName, "Failed to open input stream for file: .*");
+        assertQueryFailsIncorrectly("MERGE INTO  " + tableName + " USING (SELECT 1 a) input ON true WHEN MATCHED THEN DELETE", "Failed to open input stream for file: .*");
+        assertQueryFailsIncorrectly("TRUNCATE TABLE " + tableName, "Failed to open input stream for file: .*");
+        assertQueryFailsIncorrectly("COMMENT ON TABLE " + tableName + " IS NULL", "Failed to open input stream for file: .*");
+        assertQueryFailsIncorrectly("COMMENT ON COLUMN " + tableName + ".foo IS NULL", "Failed to open input stream for file: .*");
+        assertQueryFailsIncorrectly("CALL iceberg.system.rollback_to_snapshot(CURRENT_SCHEMA, '" + tableName + "', 8954597067493422955)", "Failed to open input stream for file: .*");
+
+        // Avoid failing metadata queries
+        assertQuery("SHOW TABLES LIKE 'test_corrupted_table_location_%' ESCAPE '\\'", "VALUES '" + tableName + "'");
+        assertQueryReturnsEmptyResult("SELECT column_name, data_type FROM information_schema.columns " +
+                "WHERE table_schema = CURRENT_SCHEMA AND table_name LIKE 'test_corrupted_table_location_%' ESCAPE '\\'");
+        assertQueryReturnsEmptyResult("SELECT column_name, data_type FROM system.jdbc.columns " +
+                "WHERE table_cat = CURRENT_CATALOG AND table_schem = CURRENT_SCHEMA AND table_name LIKE 'test_corrupted_table_location_%' ESCAPE '\\'");
+
+        assertQueryFailsIncorrectly("DROP TABLE " + tableName, "Failed to open input stream for file: .*");
+        assertThatThrownBy(() -> getQueryRunner().tableExists(getSession(), tableName))
+                .hasMessageContaining("Failed to open input stream for file:");
+    }
+
+    private String getTableLocation(String tableName)
+    {
+        return (String) computeScalar("SELECT DISTINCT regexp_replace(\"$path\", '/[^/]*/[^/]*$', '') FROM " + tableName);
+    }
+
+    private void assertQueryFailsIncorrectly(@Language("SQL") String sql, @Language("RegExp") String expectedMessageRegExp)
+    {
+        QueryRunner queryRunner = getQueryRunner();
+        Session session = getSession();
+
+        try {
+            if (queryRunner instanceof DistributedQueryRunner distributedQueryRunner) {
+                MaterializedResultWithQueryId resultWithQueryId = distributedQueryRunner.executeWithQueryId(session, sql);
+                fail(format("Expected query to fail: %s [QueryId: %s]", sql, resultWithQueryId.getQueryId()));
+            }
+            else {
+                queryRunner.execute(session, sql);
```

Comment:
This is dead code.

The `createQueryRunner()` method returns a `DistributedQueryRunner`.
```
IcebergQueryRunner.builder()
                .setMetastoreDirectory(dataDirectory)
                .build()
```